packages/core/src/shared/utilities/workspaceUtils.ts

```diff
@@ -315,6 +345,14 @@ export async function collectFiles(
         zipFilePath: string
     }[]
 > {
+    const workspaceSize = await getWorkspaceSize(sourcePaths, workspaceFolders, respectGitIgnore)
```

Comment:
Did you look for existing code that calculates workspace size? I seem to recall a few of those,  e.g. for "security scan". Please check for those and see if they can be updated to use this shared logic.